File: bots/lab-templates/ts/LabBot.ts
import { Bot, HitWallEvent, ScannedBotEvent } from "@robocode.dev/tank-royale-bot-api";

/** Lab starter — move, scan, fire. Edit & Deploy. */
class LabBot extends Bot {
  static main() {
    new LabBot().start();
  }
  override run() {
    while (this.isRunning()) {
      this.turnRadarRight(360);
      this.forward(50);
      this.turnLeft(20);
    }
  }
  override onScannedBot(_e: ScannedBotEvent) {
    this.fire(1.5);
  }
  override onHitWall(_e: HitWallEvent) {
    this.back(60);
    this.turnRight(90);
  }
}
LabBot.main();

File: bots/lab-templates/java/LabBot.java
import dev.robocode.tankroyale.botapi.*;
import dev.robocode.tankroyale.botapi.events.*;

/** Lab starter — move, scan, fire. Edit & Deploy. */
public class LabBot extends Bot {
    public static void main(String[] args) {
        new LabBot().start();
    }

    @Override
    public void run() {
        while (isRunning()) {
            turnRadarRight(360);
            forward(50);
            turnLeft(20);
        }
    }

    @Override
    public void onScannedBot(ScannedBotEvent e) {
        fire(1.5);
    }

    @Override
    public void onHitWall(HitWallEvent e) {
        back(60);
        turnRight(90);
    }
}
